protocols/imap/src/main/java/org/apache/james/imap/processor/ListProcessor.java

```diff
@@ -267,7 +267,7 @@ private List<Triple<MailboxPath, ListResponse, Optional<MailboxMetaData>>> getLi
 
         allSubscribedSearch.stream()
             .filter(subscribed -> !listRecursiveMatchPath.contains(subscribed))
-            .filter(mailboxQuery::isPathMatch)
+            .filter(searchedResultMap::containsKey)
```

Comment:
It fix issue, But I'm not sure it is a good way or not. 

The `mailboxQuery` is alway true for this case. 
`MailboxQuery{namespace=Optional[#private], user=Optional.empty, mailboxNameExpression=PrefixedRegex{prefix='', regex='*'}}`